lib/scope.js

```diff
@@ -1805,35 +1805,53 @@ module.exports = {
     /* Try to upload one or more files to a file-type input field.
     *    Does not submit, so no danger of timeout with server reload.
     */
-    
-    // Tests could be in one of two places. Figure out which directory they're in
-    // Makes an absolute path from the directory in which `npm run` was used plus our internal folders
-    let dir_source = `${ process.cwd() }/docassemble/${ session_vars.get_repo_url().match(/docassemble-(.*)/)[1] }/data/sources`;
-    let dir_features = `${ process.cwd() }/tests/features`;
-    let dir = null;
-
-    // Make sure one of the valid directories exist. This only allows
-    // one directory to be used per Step or table row. We don't anticipate
-    // future people having a `tests/features` folder, let alone both.
+
+    // Tests should be one place: docassemble/*/data/sources, where * is the only folder in `docassemble`.
+    // Makes an absolute path from the folder in which `npm run` was used plus our internal folders
+    let docassemble_pkg_path = `${ process.cwd() }/docassemble`;
+    let subpackage_name = session_vars.get_package_name().substring(12);
+    let pkg_paths = [];
     try {
-      // Try the most common directory first
-      fs.accessSync( dir_source );
-      dir = dir_source;
-      // If they have both directories, let them know which directory will be used.
-      try {
-        fs.accessSync( dir_features );
-        await scope.addToReport( scope, { type: `warning`, value: `You have both a docassemble package 'sources' folder and a 'test/features' folder. These tests will use the package's 'sources' folder.` });
-      } catch ( error_features ) { `all's well, do nothing`; }
+      fs.accessSync(docassemble_pkg_path);
+      // https://stackoverflow.com/a/24594123/11416267
+      let dirs = fs.readdirSync(docassemble_pkg_path, {withFileTypes: true}).filter(dirent => dirent.isDirectory()).map(dirent => dirent.name);
+      if (dirs.length < 1) {
+        let msg = `You can store files to upload in 'docassemble/${ subpackage_name }/data/source'. These tests could not find a folder like that starting at ${ process.cwd() }.`;
+        await scope.addToReport( scope, { type: `warning`, value: msg});
+        throw Error(msg);  // Most people will want to know the sources folder path;
+      }
+      for (let dir of dirs) {
+        pkg_paths.push(`${docassemble_pkg_path}/${dir}`);
```

Comment:
I added the ability to actually check which of the folders in `docassemble` have a `data/sources` before just blindly choosing the first one. I ran into the local issue where I had a `.mypy_cache` folder under `docassemble` as well, which would get incorrectly chosen. Minor for others, but it was very helpful for me when developing locally, and does make things a bit more resilient IMO.

Reply:
Great! Meant to comment on that. That's a great improvement.